Source/Core/VideoCommon/AVIDump.cpp

```diff
@@ -392,7 +392,10 @@ void AVIDump::CheckResolution(int width, int height)
   // (possibly width as well, but no examples known) to have a value of zero. This can occur as the
   // VI is able to be set to a zero value for height/width to disable output. If this is the case,
   // simply keep the last known resolution of the video for the added frame.
-  if ((width != s_width || height != s_height) && (width > 0 && height > 0))
+  bool width_changed = g_Config.iDumpWidth ? (s_width != g_Config.iDumpWidth) : (width != s_width);
```

Comment:
I guess we're guaranteed to get `AVIDump::Start`-ed which will reset the `s_width`. I think it's ok to keep things like this though?